docs/LwM2M_Client/Nordic/Runtime_certificate.md

```diff
@@ -0,0 +1,103 @@
+Anjay library supports secure connection between device and server. Encryption can be achieved with the use of Pre-Shared Key as it was described in [nRF9160](nRF9160.md) and [Thingy:91](Thingy91.md) pages or with certificate-based cryptography.
+
+In **Certificate Mode** an asymmetrical public-key cryptographic algorithm is used to authenticate the connection endpoints and initialize payload encryption. Appropriate certificates need to be generated for both the LwM2M Client and the LwM2M Server. Public certificates of both parties are mutually available, and each party also has access to its corresponding private key.
+
+Read more about secure communication on [Anjay's documentation](https://avsystem.github.io/Anjay-doc/BasicClient/BC-Security.html) page.
+
+## Prerequisites
+* The nRF9160 board with a USB cable.
+- Cloned [Anjay-zephyr-client](https://github.com/AVSystem/Anjay-zephyr-client#getting-started) repository
+- Installed **nrfjprog** from [Nordic Semiconductor page](https://www.nordicsemi.com/Products/Development-tools/nrf-command-line-tools/download)
+* Installed **minicom** (for Linux), **RealTerm**, **PuTTy** (for Windows), or another serial communication program.
+* An active [Coiote IoT DM](https://eu.iot.avsystem.cloud/) user account.
+
+## Build and flash the device
+0. Connect the nRF9160 board to a USB port of your machine.
+0. Go to your local **Anjay-zephyr-client** directory
+0. Set West manifest path to `Anjay-zephyr-client/demo`, manifest file to `west-nrf.yml`, and do `west update`:
+
+    ```
+    west config manifest.path Anjay-zephyr-client/demo
+    west config manifest.file west-nrf.yml
+    west update
+    ```
+
+0. Go to the **Anjay-zephyr-client/demo** folder and build a project with a runtime certificate and private key.
+
+    ```
+    west build -b <BOARD> -p -- -DCONFIG_ANJAY_ZEPHYR_RUNTIME_CERT_CONFIG=y
+    ```
+
+    where <BOARD\> should be replaced by the selected board from `boards/` directory.
+
+    !!! Note
+
+        The runtime certificate and private key do not work with Thingy:91. Other boards should work with this command.
+
+    This feature works with nrf9160dk starting from revision v0.14.0. For this board use configuration that utilizes an external flash chip and software-based cryptography:
+
+    ```
+    west build -b nrf9160dk_nrf9160_ns@0.14.0 -p -- -DCONF_FILE=prj_extflash.conf -DOVERLAY_CONFIG="overlay_nrf_mbedtls.conf"
+    ```
+
+0. Find the merged.hex file under the `build/zephyr` directory in the project folder.
+
+0. Use the nRF Connect Programmer with the downloaded `.hex` file and execute steps from the [Updating the application firmware](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_nrf9160_gs.html#updating-the-application-firmware) section.
+
+## Generate certificate
+The certificate and private key based on the SECP256R1 curve can be provided through the shell interface in `.pem` format. To generate them open terminal and use the following commands.
```

Comment:
```suggestion
The certificate and private key based on the SECP256R1 curve can be provided through the shell interface in `.pem` format. To generate them open terminal, go to the `demo/` direcotry and use the following commands.
```

Reply:
but we are in demo folder

Reply:
My bad, I would specify it explicitly that we should do it in the `demo/` directory anyway. Just for clarification for the user.